kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopEventManager.java

```diff
@@ -0,0 +1,318 @@
+        if (notification.getPath().equals(LoadManager.LOADBALANCE_BROKERS_ROOT)) {
+            this.brokersChangeHandler.handleChildChange();
+        } else if (notification.getPath().equals(getDeleteTopicsPath())) {
+            this.deletionTopicsHandler.handleChildChange();
+        }
+    }
+
+    private void getBrokers(List<String> pulsarBrokers) {
+        final Set<Node> kopBrokers = Sets.newConcurrentHashSet();
+        final AtomicInteger pendingBrokers = new AtomicInteger(pulsarBrokers.size());
+        CompletableFuture<Set<Node>> kopBrokersFuture = new CompletableFuture<>();
+        pulsarBrokers.forEach(broker -> {
+            metadataStore.get(getBrokersChangePath() + "/" + broker).whenComplete(
+                    (brokerData, e) -> {
+                        if (e != null) {
+                            log.error("Get broker {} path data failed which have an error", broker, e);
+                            kopBrokersFuture.completeExceptionally(e);
+                            return;
+                        }
+
+                        if (brokerData.isPresent()) {
+                            JsonObject jsonObject = parseJsonObject(
+                                    new String(brokerData.get().getValue(), StandardCharsets.UTF_8));
+                            JsonObject protocols = jsonObject.getAsJsonObject("protocols");
+                            JsonElement element = protocols.get("kafka");
+
+                            if (element != null) {
+                                String kopBrokerStr = element.getAsString();
+                                Node kopNode = getNode(kopBrokerStr);
+                                kopBrokers.add(kopNode);
+                            } else {
+                                if (log.isDebugEnabled()) {
+                                    log.debug("Get broker {} path currently not a kop broker, skip it.", broker);
+                                }
+                            }
+                        } else {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Get broker {} path data empty.", broker);
+                            }
+                        }
+
+                        if (pendingBrokers.decrementAndGet() == 0) {
+                            kopBrokersFuture.complete(kopBrokers);
+                        }
+                    }
+            );
+        });
+
+        kopBrokersFuture.whenComplete((brokers, e) -> {
+            if (e != null) {
+                log.error("Get pulsar brokers {} failed", pulsarBrokers, e);
+                return;
+            }
+
+            Collection<? extends Node> oldKopBrokers = adminManager.getBrokers();
+            adminManager.addBrokers(brokers);
+            log.info("Refresh kop brokers new cache {}, old brokers cache {}",
+                    adminManager.getBrokers(), oldKopBrokers);
+        });
```

Comment:
`kopBrokersFuture` is not exposed, the only usage is the callback. I think we can move this part in `metadataStore.get(getBrokersChangePath() + "/" + broker)`'s callback